### PR TITLE
feat(actuator):Adding Support for Spring Boot Actuator

### PR DESCRIPTION
**Pull request by JinQian1024**
## What is the purpose of the change
Adding Support for Spring Boot Actuator and equivalent to the QoS functions

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-spring-boot/dubbo-spring-boot-actuator/src/main/resources/META-INF/dubbo-endpoints-default.properties [hunk 1]
```diff
@@ -9,6 +9,43 @@ management.endpoint.dubboconfigs.enabled = true
...
+management.endpoint.dubboonlineinterface.enabled = true
+management.endpoint.dubbooffline.enabled = true
+management.endpoint.dubboofflineapp.enabled = true
+management.endpoint.dubboofflineinterface.enabled = true
+management.endpoint.dubbologgerinfo.enabled = true
+management.endpoint.dubboswitchlogger.enabled = true
+management.endpoint.dubboswitchloglevel.enabled = true
+management.endpoint.dubbodisabledetailprofiler.enabled = true
+management.endpoint.dubbodisablesimpleprofiler.enabled = false
+management.endpoint.dubboenabledetailprofiler.enabled = true
```
Comment (AlbumenJ) [inline dubbo-spring-boot/dubbo-spring-boot-actuator/src/main/resources/META-INF/dubbo-endpoints-default.properties]: Can we generate this automatically

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-spring-boot/dubbo-spring-boot-actuator/src/main/resources/META-INF/dubbo-endpoints-default.properties (hunk 1 above)
Comment (JinQian1024) [reply dubbo-spring-boot/dubbo-spring-boot-actuator/src/main/resources/META-INF/dubbo-endpoints-default.properties]: Perhaps I can automatically generate using the information in this file, allowing configurations to be automatically added when new QoS features are introduced. These default configuration settings can also be left empty, and if they are empty, they will be enabled by default.
image

Comment (AlbumenJ) [reply dubbo-spring-boot/dubbo-spring-boot-actuator/src/main/resources/META-INF/dubbo-endpoints-default.properties]: Yes, from this

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-spring-boot/dubbo-spring-boot-actuator/src/test/java/org/apache/dubbo/spring/boot/actuate/endpoint/DubboEndpointTest.java [hunk 2]
```diff
@@ -34,20 +34,20 @@
...
 @SpringBootTest(
-        classes = {DubboMetadataEndpoint.class},
+        classes = {DubboEndpoint.class},
         properties = {"dubbo.application.name = dubbo-demo-application"})
 @EnableAutoConfiguration
 class DubboEndpointTest {
 
     @Autowired
-    private DubboMetadataEndpoint dubboEndpoint;
+    private DubboEndpoint dubboEndpoint;
```
Comment (CrazyHZM) [inline dubbo-spring-boot/dubbo-spring-boot-actuator/src/test/java/org/apache/dubbo/spring/boot/actuate/endpoint/DubboEndpointTest.java]: Do not delete the original test case.

File: dubbo-spring-boot/dubbo-spring-boot-actuator/src/main/java/org/apache/dubbo/spring/boot/actuate/endpoint/DubboEndpoint.java [hunk 3]
```diff
@@ -16,28 +16,47 @@
...
  * @see Endpoint
- * @since 2.7.0
+ * @since 3.3.0
  */
 @Endpoint(id = "dubbo")
-public class DubboMetadataEndpoint {
+public class DubboEndpoint {
+
+    @Autowired
+    private DubboActuatorService dubboActuatorService;
```
Comment (CrazyHZM) [inline dubbo-spring-boot/dubbo-spring-boot-actuator/src/main/java/org/apache/dubbo/spring/boot/actuate/endpoint/DubboEndpoint.java]: Why not inject  `ApplicationModel`?

File: dubbo-spring-boot/dubbo-spring-boot-compatible/actuator/src/main/java/org/apache/dubbo/spring/boot/actuate/endpoint/configuration/DubboActuatorProperties.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+package org.apache.dubbo.spring.boot.actuate.endpoint.configuration;
+
+import java.util.Map;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+@Component
+@ConfigurationProperties(prefix = "management")
```
Comment (CrazyHZM) [inline dubbo-spring-boot/dubbo-spring-boot-compatible/actuator/src/main/java/org/apache/dubbo/spring/boot/actuate/endpoint/configuration/DubboActuatorProperties.java]: the prefix should be `management.endpoint` `management` affects other configurations that start with management, such as `management.server.port`

**Review by CrazyHZM — CHANGES_REQUESTED** [review]

**Review by CrazyHZM — APPROVED** [review]

**Comment by JinQian1024** [comment]
@AlbumenJ PTAL

File: dubbo-spring-boot/dubbo-spring-boot-actuator/src/main/resources/META-INF/dubbo-endpoints-default.properties [hunk 5]
```diff
@@ -2,20 +2,58 @@
...
 # "management.endpoints.web.base-path" should not be configured in this file
-
 # Re-defines path-mapping of Dubbo Web Endpoints
-
 management.endpoints.web.path-mapping.dubboshutdown = dubbo/shutdown
 management.endpoints.web.path-mapping.dubboconfigs = dubbo/configs
 management.endpoints.web.path-mapping.dubboservices = dubbo/services
 management.endpoints.web.path-mapping.dubboreferences = dubbo/references
 management.endpoints.web.path-mapping.dubboproperties = dubbo/properties
+# Set enabled for Dubbo Endpoints
```
Comment (CrazyHZM) [inline dubbo-spring-boot/dubbo-spring-boot-actuator/src/main/resources/META-INF/dubbo-endpoints-default.properties:16]: management.endpoint.dubboofflineapp.enabled
management.endpoint.dubboofflineinterface.enabled
management.endpoint.dubbogracefulshutdown.enabled
These need to default to false.

**Review by CrazyHZM — CHANGES_REQUESTED** [review]

File: dubbo-spring-boot/dubbo-spring-boot-actuator/src/main/resources/META-INF/dubbo-endpoints-default.properties [hunk 6]
```diff
@@ -2,20 +2,57 @@
...
+management.endpoint.dubbosetprofilerwarnpercent.enabled = true
+management.endpoint.dubboserializecheckstatus.enabled = true
+management.endpoint.dubboserializewarnedclasses.enabled = true
+management.endpoint.dubbodisableroutersnapshot.enabled = true
+management.endpoint.dubboenableroutersnapshot.enabled = true
+management.endpoint.dubbogetroutersnapshot.enabled = true
+management.endpoint.dubbogetrecentroutersnapshot.enabled = true
+management.endpoint.dubbopublishmetadata.enabled = true
+management.endpoint.dubbogracefulshutdown.enabled = false
+management.endpoint.dubbogetenabledroutersnapshot.enabled = true
```
Comment (AlbumenJ) [inline dubbo-spring-boot/dubbo-spring-boot-actuator/src/main/resources/META-INF/dubbo-endpoints-default.properties]: Should only enable `live`, `ready` and `startup` by default.

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-spring-boot/dubbo-spring-boot-compatible/actuator/src/main/java/org/apache/dubbo/spring/boot/actuate/endpoint/configuration/DubboEndpointPropertiesGenerator.java [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+                    }
+                    writer.newLine();
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    // 根据提取的关键字生成新的配置
```
Comment (AlbumenJ) [inline dubbo-spring-boot/dubbo-spring-boot-compatible/actuator/src/main/java/org/apache/dubbo/spring/boot/actuate/endpoint/configuration/DubboEndpointPropertiesGenerator.java]: Comment in English~

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Review by CrazyHZM — APPROVED** [review]

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**